src/components/sidebar/helpers/add-nav-item-meta-data.ts

```diff
@@ -0,0 +1,57 @@
+import {
```

Comment:
I plan to add more comments into this file if we decide the approach is solid. But no functionality should be changed!

Reply:
I think I'll actually wait to do this, since some of these interfaces will move soon!

Reply:
Added in https://github.com/hashicorp/dev-portal/pull/326/commits/ffa281866bea1e3bc95050ae58a190c3d7009750